pkg/cmd/server/start/start_node.go

```diff
@@ -249,30 +279,46 @@ func (o NodeOptions) IsRunFromConfig() bool {
 	return (len(o.ConfigFile) > 0)
 }
 
-func StartNode(nodeConfig configapi.NodeConfig) error {
+func StartNode(nodeConfig configapi.NodeConfig, components *utilflags.ComponentFlag) error {
 	config, err := kubernetes.BuildKubernetesNodeConfig(nodeConfig)
 	if err != nil {
 		return err
 	}
-	glog.Infof("Starting node %s (%s)", config.KubeletServer.HostnameOverride, version.Get().String())
+
+	if components.Enabled(ComponentKubelet) {
+		glog.Infof("Starting node %s (%s)", config.KubeletServer.HostnameOverride, version.Get().String())
+	} else {
+		glog.Infof("Starting node networking %s (%s)", config.KubeletServer.HostnameOverride, version.Get().String())
+	}
+
+	_, kubeClientConfig, err := configapi.GetKubeClient(nodeConfig.MasterKubeConfig)
+	if err != nil {
+		return err
+	}
+	glog.Infof("Connecting to API server %s", kubeClientConfig.Host)
 
 	// preconditions
-	config.EnsureVolumeDir()
-	config.EnsureDocker(docker.NewHelper())
-
-	// async starts
-	config.RunKubelet()
-	config.RunSDN()
-	config.RunProxy()
-
-	// HACK: RunProxy resets bridge-nf-call-iptables from what openshift-sdn requires
-	if config.SDNPlugin != nil {
-		sdnPluginName := nodeConfig.NetworkConfig.NetworkPluginName
-		if sdnPluginName == osdn.SingleTenantPluginName() || sdnPluginName == osdn.MultiTenantPluginName() {
-			if err := sysctl.SetSysctl("net/bridge/bridge-nf-call-iptables", 0); err != nil {
-				glog.Warningf("Could not set net.bridge.bridge-nf-call-iptables sysctl: %s", err)
-			}
-		}
+	if components.Enabled(ComponentKubelet) {
+		config.EnsureKubeletAccess()
+		config.EnsureVolumeDir()
+		config.EnsureDocker(docker.NewHelper())
+	}
+
+	// SDN plugins get the opportunity to filter service rules, so they start first
+	if components.Enabled(ComponentPlugins) {
+		config.RunPlugin()
```

Comment:
That's the config.  You passed "openshift start node --config=....
--enable=plugins", enable plugins does the right thing and stays open.

On Fri, Feb 19, 2016 at 11:57 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/server/start/start_node.go
> <https://github.com/openshift/origin/pull/6920#discussion_r53486726>:
>
> > -	if config.SDNPlugin != nil {
> > -		sdnPluginName := nodeConfig.NetworkConfig.NetworkPluginName
> > -		if sdnPluginName == osdn.SingleTenantPluginName() || sdnPluginName == osdn.MultiTenantPluginName() {
> > -			if err := sysctl.SetSysctl("net/bridge/bridge-nf-call-iptables", 0); err != nil {
> > -				glog.Warningf("Could not set net.bridge.bridge-nf-call-iptables sysctl: %s", err)
> > -			}
> > -		}
> > +	if components.Enabled(ComponentKubelet) {
> > +		config.EnsureKubeletAccess()
> > +		config.EnsureVolumeDir()
> > +		config.EnsureDocker(docker.NewHelper())
> > +	}
> > +
> > +	// SDN plugins get the opportunity to filter service rules, so they start first
> > +	if components.Enabled(ComponentPlugins) {
> > +		config.RunPlugin()
>
> what does this do if you do start node --enable=plugins and don't
> actually have a sdn plugin configured? does the process hang in the
> foreground or tell you you don't have any plugins?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6920/files#r53486726>.
>